src/ompl/geometric/planners/rrt/src/RRTstar.cpp

```diff
@@ -101,6 +129,16 @@ void ompl::geometric::RRTstar::setup()
         {
             OMPL_INFORM("%s: No optimization objective specified. Defaulting to optimizing path length for the allowed planning time.", getName().c_str());
             opt_.reset(new base::PathLengthOptimizationObjective(si_));
+
+            if (opt_->hasCostToGoHeuristic() == false)
+            {
+                OMPL_INFORM("%s: No cost-to-go heuristic set. Defaulting to the distance from the state to the goal.", getName().c_str());
+                //Bind to the goalRegionCostToGo function.
+                opt_->setCostToGoHeuristic( boost::bind(&base::goalRegionCostToGo, _1, _2) );
```

Comment:
Should we not move this code as part of setup() for the OptimizationObjective? Or at least for the ProblemDefinition?